### PR TITLE
Make ready for Pimcore X

### PR DESCRIPTION
**Pull request by ITspirit**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
👏🙌 will review in the upcoming week

**Comment by dpfaffenbauer** [comment]
Todos:
 - Symfony 5 Directories
 - File Based Readers and Exporters: Use Flysystem
 - Tests with Github Actions

**Comment by dpfaffenbauer** [comment]
done, might anybody check? @aarongerig maybe? @dkarlovi please too?

**Comment by aarongerig** [comment]
@dpfaffenbauer Sorry, it's currently somewhat difficult due to lack of capacity. As soon as I find some spare time, I will try to test this.

**Comment by dpfaffenbauer** [comment]
@aarongerig time this week? Otherwise this will get merged without your review

**Review by dpfaffenbauer — APPROVED** [review]

**Comment by aarongerig** [comment]
@dpfaffenbauer I'll try, otherwise go for it!

File: composer.json [hunk 1]
```diff
@@ -25,26 +29,26 @@
...
-    "coreshop/resource-bundle": "^2.0 | ^3.0",
-    "coreshop/rule-bundle": "^2.0 | ^3.0",
-    "coreshop/pimcore-bundle": "^2.0 | ^3.0",
-    "coreshop/pimcore": "^2.0 | ^3.0",
-    "pimcore/pimcore": "^6.5",
-    "league/csv": "^9.0",
-    "box/spout": "^2.7"
+    "php": ">=8.0",
+    "ext-json": "*",
+    "box/spout": "^3.0",
```
Comment (aarongerig) [inline composer.json:34]: This dependency update (from `^2.7` to `^3.0`) doesn't go well with the Pimcore CMF (see https://github.com/pimcore/customer-data-framework/blob/master/composer.json#L9), but IMO they should update their dependency as well.

Comment (ITspirit) [reply composer.json:34]: There is already a pull request for the upcoming CMF Version 3.1

Comment (aarongerig) [reply composer.json:34]: Ah right, didn't notice that. Even better. 👍🏻

**Comment by aarongerig** [comment]
@ITspirit When I try to run the import command I get following error:
```bash
In ImportCommand.php line 80:
Warning: Undefined array key "userId"
```

It would be probably best to use an `isset()` check there.

**Comment by aarongerig** [comment]
@ITspirit When defining a localized field as primary key, an exception within `doctrine\dbal` is thrown: e.g. `column 'title~de' not found in where clause`.

Not sure though if that ever was supported, @dpfaffenbauer ?

**Comment by aarongerig** [comment]
@ITspirit Another one I found is the following. I used the "Custom" field to insert a default value (with default_value interpreter) into `o_published`. This doesn't work and throws following error:

```bash
ErrorException: Warning: Undefined array key "custom" in /var/www/html/dev/DataDefinitions/src/DataDefinitionsBundle/Importer/Importer.php:355
```

**Comment by dpfaffenbauer** [comment]
> @ITspirit When defining a localized field as primary key, an exception within `doctrine\dbal` is thrown: e.g. `column 'title~de' not found in where clause`.
> 
> Not sure though if that ever was supported, @dpfaffenbauer ?

no, was never supported

**Comment by dkarlovi** [comment]
Sorry, just can't manage to find time to review here. 🙈

**Comment by dpfaffenbauer** [comment]
@aarongerig fixed your findings

**Comment by dpfaffenbauer** [comment]
this was just merged, but github did not realize it....

**Pull request closed without merging**